glean-core/src/core/mod.rs

```diff
@@ -758,19 +758,40 @@ impl Glean {
             .get_value(self, None)
     }
 
-    /// Set configuration to override the default metric enabled/disabled state, typically from a
+    /// Set configuration to override the default state, typically initiated from a
     /// remote_settings experiment or rollout
     ///
     /// # Arguments
     ///
-    /// * `json` - The stringified JSON representation of a `MetricsEnabledConfig` object
-    pub fn set_metrics_enabled_config(&self, cfg: MetricsEnabledConfig) {
-        // Set the current MetricsEnabledConfig, keeping the lock until the epoch is
+    /// * `json` - The stringified JSON representation of a `RemoteSettingsConfig` object
+    pub fn set_remote_settings_config(&self, cfg: RemoteSettingsConfig) {
+        // Set the current RemoteSettingsConfig, keeping the lock until the epoch is
         // updated to prevent against reading a "new" config but an "old" epoch
```

Comment:
_software architect hat on_ To me this suggests that the lock should cover both the epoch and the config, since we want to forbid using one without the other. Might be a follow-up.